### DISCUSSION TITLE
Ability to update of delete the built-in clients

### DISCUSSION THREAD

**Discussion by ossdhaval** [JanssenProject/jans#10239] [Ideas]
Out of the box, the Janssen Server creates clients that are used or should be used exclusively by the Janssen Server  modules. See below:

![image](https://github.com/user-attachments/assets/c4963b31-1b09-4bd1-aa81-0712e87b640c)

All these clients can be updated and deleted by the server admin. This can also happen by mistake. Doing so will leave the Janssen Server impaired as the corresponding module may not work properly. 

Proposal: Updating/Deleting action on built-in clients should either be disabled or there should be warning  displayed before moving forward.

**Comment by yuriyz** [comment]
We need to add confirmation dialog where we should inform admin that deletion may disrupt functionality.

**Comment by nynymike** [comment]
Yes, it's implicit that if you delete these clients you're going to be in trouble :-)  What is the default expiration? Maybe it should be never...

**Reply by yuriyz** [reply]
It's non-expirable. We have `del: false` for built-in clients (which stands for deletable=false in code).

**Comment by ossdhaval** [comment]
@yuriyz @nynymike @devrimyatar 

So, net-net, do we all agree that we should add a warning screen before deleting such a client?

**Reply by yuriyz** [reply]
yes from me

**Comment by manojs1978** [comment]
we are getting warning before deleting client ,for user created and inbuild client

<img width="1366" height="768" alt="Screenshot From 2026-02-05 18-18-20" src="https://github.com/user-attachments/assets/940bb517-1fe6-4d76-8c71-3296e0405a37" />
 
<img width="1366" height="768" alt="Screenshot From 2026-02-05 18-18-34" src="https://github.com/user-attachments/assets/00f954eb-c97b-440e-943b-78f379b1d7fc" />

**Comment by moabu** [comment]
This is already implemented.
